src/Datasource/ConnectionRegistry.php

```diff
@@ -65,20 +65,13 @@ protected function _throwMissingClassError(string $class, ?string $plugin): void
      *
      * Part of the template method for Cake\Core\ObjectRegistry::load()
      *
-     * If a callable is passed as first argument, The returned value of this
-     * function will be the result of the callable.
-     *
-     * @param \Cake\Datasource\ConnectionInterface|callable|string $class The classname or object to make.
+     * @param \Cake\Datasource\ConnectionInterface|string $class The classname or object to make.
      * @param string $alias The alias of the object.
      * @param array $config An array of settings to use for the datasource.
      * @return \Cake\Datasource\ConnectionInterface A connection with the correct settings.
      */
-    protected function _create($class, string $alias, array $config)
+    protected function _create(object|string $class, string $alias, array $config): ConnectionInterface
```

Comment:
Not all connection classes implement `ConnectionInterface`, for e.g. https://github.com/UseMuffin/Webservice/blob/9bde1646d6ed938a5e2b2a0dd6e78b3a89b6ac2a/src/Connection.php#L16.

It's because `ConnectionInterface` has methods like `transactional()` which are irrelevant for non RDBMS datasource.

Reply:
This is the big issue. I made changes so everything passes since it expects ConnectionInterface. You can see how FakeConnection was failing as well.

I'm not sure ConnectionRegistry can or should return random objects.

Reply:
Well currently it can return any object. Let's limit the scope of this PR to only add types as per current behavior (unless it's a trivial change).

We can discuss changes to `ConnectionRegistry`/`ConnectionInterface` separately.

Reply:
Yup, it has to be just `object` return type. 

What about `callable` support? Is that necessary? I think we can expand `object|string` to `callable|object|string` now that I think about it. We cannot restrict `object|string` to just `string` though.

Reply:
I think only connection registry has callable support to theoretically allow lazy creation of connection instance. That way connecting to a (external) datasource would be avoided unless needed.

To maintain the status quo we could just use `callable|object|string`.